csharp/lib/src/lib.rs

```diff
@@ -26,37 +31,96 @@ pub struct Client {
     runtime: Runtime,
 }
 
-fn create_connection_request(
-    host: String,
-    port: u32,
-    use_tls: bool,
+/// Convert raw C string to a rust string.
+/// # Safety
+/// Unsafe function because creating a string from a pointer.
```

Comment:
Safety documentation needs to communicate to the caller of the function what they need to do to ensure that they are not passing arguments that would cause undefined behaviour. Please fix throughout.

Also, I prefer formatting using bullet points similar to the official docs (example: https://doc.rust-lang.org/src/alloc/vec/mod.rs.html#587)

Reply:
Copied from your PR and adopted